core/src/test/scala/unit/kafka/server/BaseRequestTest.scala

```diff
@@ -136,7 +135,7 @@ abstract class BaseRequestTest extends IntegrationTestHarness {
   }
 
   def sendWithHeader(request: AbstractRequest, header: RequestHeader, socket: Socket): Unit = {
-    val serializedBytes = Utils.toArray(RequestTestUtils.serializeRequestWithHeader(header, request))
+    val serializedBytes = Utils.toArray(request.serializeWithHeader(header))
     sendRequest(socket, serializedBytes)
```

Comment:
It seems this method can be replaced by `IntegrationTestUtils.sendRequest`

Reply:
This is a great suggestion! I even found that the whole `BaseRequestTest` is a duplicate of `IntegrationTestUtils`, how about open another PR for this refactor since this would change a lot of classes and the `IntegrationTestUtils` is under `integration` package while `BaseRequestTest` is under `unit` package.

Reply:
sounds good to me